src/contracts/core/AllocationManager.sol

```diff
@@ -0,0 +1,867 @@
+    /**
+     * @notice Called by the delagation manager to set delay when operators register.
+     * @param operator The operator to set the delay on behalf of.
+     * @param delay The allocation delay in seconds.
+     * @dev msg.sender is assumed to be the delegation manager.
+     */
+    function setAllocationDelay(address operator, uint32 delay) external {
+        require(msg.sender == address(delegation), OnlyDelegationManager());
+        _setAllocationDelay(operator, delay);
+    }
+
+    /**
+     * @notice Called by operators to set their allocation delay.
+     * @param delay the allocation delay in seconds
+     * @dev msg.sender is assumed to be the operator
+     */
+    function setAllocationDelay(
+        uint32 delay
+    ) external {
+        require(delegation.isOperator(msg.sender), OperatorNotRegistered());
+        _setAllocationDelay(msg.sender, delay);
+    }
+
+    /**
+     * @notice For all pending deallocations that have become completable, their pending free magnitude can be
+     * added back to the free magnitude of the (operator, strategy) amount. This function takes a list of strategies
+     * and adds all completable deallocations for each strategy, updating the freeMagnitudes of the operator
+     *
+     * @param operator address to complete deallocations for
+     * @param strategies a list of strategies to complete deallocations for
+     * @param numToComplete a list of number of pending free magnitude deallocations to complete for each strategy
+     *
+     * @dev can be called permissionlessly by anyone
+     */
+    function updateFreeMagnitude(
+        address operator,
+        IStrategy[] calldata strategies,
+        uint16[] calldata numToComplete
+    ) external {
+        require(strategies.length == numToComplete.length, InputArrayLengthMismatch());
+        require(delegation.isOperator(operator), OperatorNotRegistered());
+        for (uint256 i = 0; i < strategies.length; ++i) {
+            _updateFreeMagnitude({operator: operator, strategy: strategies[i], numToComplete: numToComplete[i]});
+        }
+    }
+
+    /**
+     * @notice Modifies the propotions of slashable stake allocated to a list of operatorSets for a set of strategies
+     * @param operator address to modify allocations for
+     * @param allocations array of magnitude adjustments for multiple strategies and corresponding operator sets
+     * @param operatorSignature signature of the operator if msg.sender is not the operator
+     * @dev Updates freeMagnitude for the updated strategies
+     * @dev Must be called by the operator or with a valid operator signature
+     * @dev For each allocation, allocation.operatorSets MUST be ordered in ascending order according to the
+     * encoding of the operatorSet. This is to prevent duplicate operatorSets being passed in. The easiest way to ensure
+     * ordering is to sort allocated operatorSets by address first, and then sort for each avs by ascending operatorSetIds.
```

Comment:
nit: `sort by address first` should probably specify that this is the AVS address

Reply:
This require has been removed and natspec since has been updated